partner_event/models/event_registration.py

```diff
@@ -35,19 +36,17 @@ def create(self, vals):
             attendee_partner = Partner.search([("email", "=ilike", email)], limit=1)
             event = Event.browse(vals["event_id"])
             if attendee_partner:
-                vals["name"] = vals.setdefault("name", attendee_partner.name)
-                vals["phone"] = vals.setdefault("phone", attendee_partner.phone)
+                vals["name"] = attendee_partner.name
```

Comment:
Sorry, but the change is incorrect, as it changes the expected behavior. For example, for the case that the same email address is used for several attendees (a real possibility when  the registration is made from the website) we'd loose that information.

Do these keys come in the `vals` dictionary when creating the attendee. If so, then use coercion to default to the proper value. Something like

```python
vals["name"] = not vals.get("name") and attendee_partner.name or vals.get("name")
```